### PR TITLE
cleanup: use backoff v4

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pkg/istio-agent/agent.go [hunk 1]
```diff
@@ -258,14 +258,12 @@ func (a *Agent) initializeEnvoyAgent() error {
 					break
 				}
 				request.sent = false
-				delay := time.Duration(rand.Int()%backoff) * time.Millisecond
+				delay := policy.NextBackOff()
+				if delay == policy.Stop {
+					log.Fatal("exhausted xDS request backoff budget of 15 minutes")
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply pkg/istio-agent/agent.go:263]: That's clearly not the correct behavior, we should fix that if so. There was a PR fixing this, not sure if it was incorrect or just didn't address all usages? I think once you hit the max you should stay at the max backoff time? I think that's what envoy does.

[the remark this answers is not in the document]
Comment (kyessenov) [reply pkg/istio-agent/agent.go:263]: We must continue jittering after max. Not sure if the library does it though.

**Review by howardjohn — APPROVED** [review]

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by kyessenov** [comment]
@bianpengyuan  Need P&T approval.

**Comment by kyessenov** [comment]
@douglas-reid can I get your approval for this cleanup?

**Pull request closed without merging**